plugins/BEdita/API/src/Controller/LoginController.php

```diff
@@ -207,10 +218,35 @@ public function update()
         $this->set('_serialize', ['entity']);
     }
 
+    /**
+     * Check current password if a password change is requested.
+     * If `password` is set in requesta data current valid password must be in `old_password`
+     *
+     * @param \BEdita\Core\Model\Entity\User $entity Logged user entity.
+     * @param array $data Request data.
+     * @throws \Cake\Network\Exception\BadRequestException Throws an exception if current password is not correct.
+     * @return void
+     */
+    protected function checkPassword(User $entity, array $data)
+    {
+        if (!isset($data['password'])) {
+            return;
+        }
+
+        if (empty($data['old_password'])) {
+            throw new BadRequestException(__('Missing current password'));
+        }
+
+        $hasher = PasswordHasherFactory::build(self::PASSWORD_HASHER);
```

Comment:
Instead of explicitly re-building the hasher, we might access the one used by `FormAuthenticate`:

```php
$hasher = $this->Auth->getAuthenticate('Form')->passwordHasher();
if (!$hasher->check($data['old_password'], $entity->password_hash)) {
    throw new BadRequestException('You\'re basically a bad person');
}
```

This way we could avoid the class constant.

Reply:
In this context `FormAuthenticate` is not initialized, but only in `login` action.
If we need to re-initialize it we could probably keep my version... :thinking: